### PR TITLE
feat: support deposit/withdraw vault flow

### PR DESCRIPTION
**Pull request by sosaucily**
This pull request updates the website to align with the new Ethereum smart contract structure and flow. It includes updates to dlc-btc-lib to enable the necessary functions for supporting deposit and withdrawal processes. The UI is modified to reflect the updated flow, with text adjustments to match the new processes, and renaming "Redeem" to "Withdraw."

### CODE REVIEW

File: src/app/components/mint-unmint/components/unmint/components/unmint-vault-selector.tsx [hunk 1]
```diff
@@ -40,64 +46,64 @@ export function UnmintVaultSelector({
...
-  async function handleUnmint(): Promise<void> {
+  async function handleBurn(withdrawAmount: number): Promise<void> {
     if (selectedVault) {
       try {
         setIsSubmitting(true);
-        const currentRisk = await fetchRisk();
-        if (currentRisk === 'High') throw new Error('Risk Level is too high');
-        await closeVault(selectedVault.uuid);
+        // const currentRisk = await fetchRisk();
+        // if (currentRisk === 'High') throw new Error('Risk Level is too high');
```
Comment (sosaucily) [inline src/app/components/mint-unmint/components/unmint/components/unmint-vault-selector.tsx:54]: Was this a temporarily comment, or do you want this gone?

Btw, is there any way to verify in mainnet that the risk_analysis is working? I guess we can check the networks tab and make sure it's making a request, and check the response?

Please make sure to log something to the console.error if that request fails for some reason

File: src/app/components/mint-unmint/components/unmint/components/unmint-vault-selector.tsx [hunk 2]
```diff
@@ -40,64 +46,64 @@ export function UnmintVaultSelector({
...
+        await withdrawVault(selectedVault.uuid, formattedWithdrawAmount);
+        await getVault(selectedVault.uuid, VaultState.FUNDED).then(() => {
+          dispatch(mintUnmintActions.setUnmintStep([1, selectedVault.uuid]));
+        });
       } catch (error) {
         setIsSubmitting(false);
-        throw new Error('Error closing vault');
+        toast({
+          title: 'Failed to sign Transaction',
+          description: error instanceof Error ? error.message : '',
```
Comment (sosaucily) [inline src/app/components/mint-unmint/components/unmint/components/unmint-vault-selector.tsx:64]: is error ever not an instance of `Error`?

File: src/app/components/mint-unmint/components/walkthrough/walkthrough.tsx [hunk 3]
```diff
@@ -120,6 +96,40 @@ export function Walkthrough({ flow, currentStep }: WalkthroughProps): React.JSX.
...
+        //     <WalkthroughLayout>
+        //       <WalkthroughHeader
+        //         currentStep={currentStep}
+        //         title={'Mint dlcBTC'}
+        //         blockchain={'ethereum'}
+        //       />
+        //       <Text color={'white.01'} fontSize={'sm'}>
+        //         Wait for Bitcoin to get locked on chain{' '}
+        //         <Link
+        //           color={'accent.lightBlue.01'}
```
Comment (sosaucily) [inline src/app/components/mint-unmint/components/walkthrough/walkthrough.tsx:110]: don't hesitate to delete :) github will remember for you

**Review by sosaucily — COMMENTED** [review]
approved, but not letting me approve :)

**Review by Polybius93 — APPROVED** [review]

**Pull request merged**